### PR TITLE
Add uap10.1 target to System.Runtime.WindowsRuntime

### PR DESCRIPTION
**Pull request by luqunl**
1. use BufferStreamWrapper to expose two internal properties. 
2. remove netcore50aot related TargetGroup
3. remove some place using #if netstandard due to netcore50aot API
4. add typeforward for Windows.Foundation.Size/Rect/point

CC: @jkotas, @weshaggard , @yizhang82 , @tarekgh , @SedarG, @ianhays 
currently also update netstandard to use BufferStreamWrapper instead. if you guys has concern about this, please let me know-- I can add the following code instead.
#if netstandard 
   use BufferStream
#else
   use BufferStreamWrapper
#endif

### CODE REVIEW

File: src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj [hunk 1]
```diff
@@ -45,6 +41,38 @@
...
+      <Assembly>System.Private.Interop</Assembly>
+    </SeedTypePreference>
+    <SeedTypePreference Include="Windows.Foundation.Size">
+      <Assembly>System.Private.Interop</Assembly>
+    </SeedTypePreference>
+  </ItemGroup>
+  <ItemGroup Condition="'$(TargetGroup)' == 'uap101aot'">
+    <TargetingPackReference Include="System.Private.Interop" />
+    <TargetingPackReference Include="System.Private.Corelib" />
+    <TargetingPackReference Include="System.Private.Threading" />
```
Comment (jkotas) [inline src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]: You should not need `System.Private.Threading` anymore.

File: src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj [hunk 2]
```diff
@@ -54,9 +82,11 @@
     <Compile Include="$(CommonPath)\Interop\Windows\mincore\Interop.GetRestrictedErrorInfo.cs">
       <Link>Common\Interop\Windows\mincore\Interop.GetRestrictedErrorInfo.cs</Link>
     </Compile>
+    <Compile Include="System\Windows\Point.cs" />
+    <Compile Include="System\Windows\Rect.cs" />
+    <Compile Include="System\Windows\Size.cs" />
   </ItemGroup>
-  <ItemGroup Condition="'$(TargetGroup)'=='netcore50aot'">
-    <Compile Include="System\IO\BufferedStream.cs" />
```
Comment (jkotas) [inline src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]: Could you please delete `BufferedStream.cs` since it is unused file after your change?

File: src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj [hunk 3]
```diff
@@ -9,26 +9,22 @@
...
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <ClsCompliant>true</ClsCompliant>
+    <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
     <!-- CS1698 - Disable warning about reference to 4.0.0.0 System.Runtime.WindowsRuntime having same simple name as target assembly -->
     <NoWarn>$(NoWarn)1698</NoWarn>
     <ProjectGuid>{844A2A0B-4169-49C3-B367-AFDC4894E487}</ProjectGuid>
     <PackageTargetRuntime>win8</PackageTargetRuntime>
-    <PackageTargetRuntime Condition="'$(TargetGroup)' == 'netcore50aot'">win8-aot</PackageTargetRuntime>
+    <PackageTargetRuntime Condition="'$(TargetGroup)' == 'uap101aot'">win8-aot</PackageTargetRuntime>
     <NuGetTargetMoniker Condition="'$(TargetGroup)' == ''">.NETStandard,Version=v1.3</NuGetTargetMoniker>
```
Comment (jkotas) [inline src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]: This needs to be `.NETStandard,Version=1.7`. I have just merged commit that made this change for other reason - be careful to resolve the conflict correctly.

File: src/System.Runtime.WindowsRuntime/src/project.json [hunk 4]
```diff
@@ -8,6 +8,12 @@
       "imports": [
         "dotnet5.4"
       ]
+    },
+    "uap10.1": {
```
Comment (jkotas) [inline src/System.Runtime.WindowsRuntime/src/project.json]: Also delete `src\System.Runtime.WindowsRuntime\src\netcore50aot\project.json` and remove references to it from the `.csproj` file.

File: src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj (hunk 3 above)
Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]: Merged this change to use  .NETStandard,Version=1.7

File: src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj (hunk 1 above)
Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]: System.Runtime.WindowRuntime need to use Progress<T> in System.Private.Threading

File: src/System.Runtime.WindowsRuntime/src/project.json (hunk 4 above)
Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/project.json]: deleted this file. I don't see it reference netcore50aot\project.json  in .csproj file

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+namespace System.IO
+{
+    /// <summary>
+    /// A BufferedStream wrapper which expose BufferedStream's UnderlyingStream and BufferSize
+    /// </summary>
+    internal sealed class BufferedStreamWrapper : Stream
+    {
+        private BufferedStream _bufferedStream;
+        private Stream _stream;                               // Underlying stream.  Close sets _stream to null.
+        private readonly Int32 _bufferSize;                   // Length of internal buffer (not counting the shadow buffer).
```
Comment (justinvp) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Nit: The corefx coding style is to use language keywords instead of BCL types, so `Int32`, `Int64`, `Byte`, `String`, etc. should be `int`, `long`, `byte`, `string`, etc. Applies throughout.

Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Thanks for your comments. Fixed and updated the change

Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Thanks for your comments. Fixed and updated the change

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.WindowsRuntime.Internal;
+using System.Threading;
+using System.Collections.ObjectModel;
+using System.Security;
+using System.Threading.Tasks;
```
Comment (justinvp) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Nit: "Remove and Sort" usings

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+            _bufferedStream.Flush();
+        }
+
+        public override Task FlushAsync(CancellationToken cancellationToken) 
+        {
+             EnsureNotClosed();
+             return _bufferedStream.FlushAsync();
+        }
+
+        public override IAsyncResult BeginRead(Byte[] buffer, int offset, int count, AsyncCallback callback, Object state)
```
Comment (justinvp) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Nit: `Byte[]` -> `byte[]` and `Object` -> `object`

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+            return _bufferedStream.ReadByte();
+        }
+
+        public override void Write(Byte[] array, int offset, int count)
+        {
+            EnsureNotClosed();
+            _bufferedStream.Write(array, offset, count);
+        }
+
+        public override IAsyncResult BeginWrite(Byte[] buffer, int offset, int count, AsyncCallback callback, Object state)
```
Comment (justinvp) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Nit: `Byte[]` -> `byte[]` and `Object` -> `object`

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+            return _bufferedStream.BeginWrite(buffer, offset, count, callback, state);
+        }
+
+        public override void EndWrite(IAsyncResult asyncResult)
+        {
+           EnsureNotClosed();
+            _bufferedStream.EndWrite(asyncResult);
+        }
+
+        public override Task WriteAsync(Byte[] buffer, int offset, int count, CancellationToken cancellationToken)
```
Comment (justinvp) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Nit: `Byte[]` -> `byte[]`

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+        {
+            EnsureNotClosed();
+            return _bufferedStream.WriteAsync(buffer, offset, count, cancellationToken);
+        }
+
+        public override void WriteByte(Byte value)
```
Comment (justinvp) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Nit: `Byte` -> `byte`

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs (hunk 7 above)
Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: oh. forgot these two. Fixed and updated.

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+            return _bufferedStream.EndRead(asyncResult);
+        }
+
+        public override int Read(byte[] array, int offset, int count)
+        {
+            EnsureNotClosed();
+            return _bufferedStream.Read(array, offset, count);
+        }
+
+        public override Task<int> ReadAsync(Byte[] buffer, int offset, int count, CancellationToken cancellationToken)
```
Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Fixed and updated.

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+            return _bufferedStream.ReadAsync(buffer, offset, count, cancellationToken);
+        }
+
+        public override int ReadByte()
+        {
+            EnsureNotClosed();
+            return _bufferedStream.ReadByte();
+        }
+
+        public override void Write(Byte[] array, int offset, int count)
```
Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Fixed and updated.

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs (hunk 8 above)
Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Fixed and updated.

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs (hunk 9 above)
Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Fixed and updated.

File: src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs (hunk 10 above)
Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Fixed and updated.

Comment (luqunl) [inline src/System.Runtime.WindowsRuntime/src/System/IO/BufferedStreamWrapper.cs]: Thanks for this. also created an issue for some other files under System.Runtime.WindowsRuntime doesn't use this code style.

https://github.com/dotnet/corefx/issues/12384

**Comment by jkotas** [comment]
Unix builds are failing with:

`The partial facade assembly's name, version, and public key token must exactly match the contract to be filled. Contract: Assembly(Name=System.Runtime.WindowsRuntime, Version=4.0.13.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a), Facade: Assembly(Name=System.Runtime.WindowsRuntime, Version=4.0.13.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)`

File: src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj [hunk 13]
```diff
@@ -5,30 +5,25 @@
   </PropertyGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <PropertyGroup>
-    <UseECMAKey>true</UseECMAKey>
+    <UseECMAKey Condition="'$(UseECMAKey)' == ''">true</UseECMAKey>
```
Comment (weshaggard) [inline src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]: Undo this change and in the ref/System.Runtime.WindowsRuntime.csproj you need to remove the condition so it matches and always uses the ECMA key. That should fix your non-windows build issues.

**Comment by weshaggard** [comment]
```
/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_release_prtest/src/Common/src/Interop/Windows/mincore/Interop.GetRestrictedErrorInfo.cs(14,32): error CS0122: 'IRestrictedErrorInfo' is inaccessible due to its protection level [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_release_prtest/src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]
14:26:32 System/IO/BufferedStreamWrapper.cs(33,18): error CS0122: 'BufferedStream' is inaccessible due to its protection level [/mnt/j/workspace/dotnet_corefx/master/ubuntu14.04_release_prtest/src/System.Runtime.WindowsRuntime/src/System.Runtime.WindowsRuntime.csproj]
```

Those errors are because those are internal APIs in System.Private.CoreLib. For better or worse System.Runtime.WindowsRuntime has InternalsVisibleTo System.Private.CoreLib which normally gives it access to those types. However on non-windows builds you not using the ECMA key for this library so that breaks the InternalsVisibleTo as it matches on full assembly identity. 

To fix this undo your change https://github.com/dotnet/corefx/pull/12373/files#diff-f300755f241fd4066230fd43d98b6f1eR8 and remove the condition in ref as well.

As an aside this points out that we are using the BufferedStream implementation in System.Private.CoreLib on CoreCLR which we should fix to have it use the one from System.IO as well by adding a reference to it.

**Comment by luqunl** [comment]
@weshaggard , yes. after I updated them(source project file and ref project file) to always use ECMA key -- CI pass.

**Comment by luqunl** [comment]
@weshaggard, about switch to use BufferedStream from System.Runtime, I can create another PR instead.
@SedarG, use another PR to solve this version problem
